### PR TITLE
Fix `output_size` dynamic dimensions support for `Col2Im-15` for CPU `StaticShape`

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details
   - `trace` case has been fixed, `export` case is still failing for PT FE affected models, but now with an error unrelated to `Col2Im`

### Tickets:
 - CVS-148673

### CODE REVIEW

File: tests/model_hub_tests/pytorch/timm_models [hunk 1]
```diff
@@ -532,11 +532,11 @@ vit_wee_patch16_reg1_gap_256.sbb_in1k,None
...
-volo_d1_224.sail_in1k,None,xfail,Cannot get length of dynamic dimension
-volo_d2_224.sail_in1k,None,xfail,Cannot get length of dynamic dimension
-volo_d3_224.sail_in1k,None,xfail,Cannot get length of dynamic dimension
-volo_d4_224.sail_in1k,None,xfail,Cannot get length of dynamic dimension
-volo_d5_224.sail_in1k,None,xfail,Cannot get length of dynamic dimension
+volo_d1_224.sail_in1k,None
+volo_d2_224.sail_in1k,None
+volo_d3_224.sail_in1k,None
+volo_d4_224.sail_in1k,None
+volo_d5_224.sail_in1k,None
```
Comment (mvafin) [inline tests/model_hub_tests/pytorch/timm_models:539]: You also should be able to remove xfail from this one: https://github.com/openvinotoolkit/openvino/blob/bc5e90200708148e297f5b3054e7e64e74ca2ed5/tests/model_hub_tests/pytorch/hf_transformers_models#L130

Comment (mvafin) [reply tests/model_hub_tests/pytorch/timm_models:539]: Please also add one volo model in precommit here: https://github.com/openvinotoolkit/openvino/blob/bc5e90200708148e297f5b3054e7e64e74ca2ed5/tests/model_hub_tests/pytorch/test_timm.py#L74

No need to remove any other model, just add, timm models are usually relatively fast.

Comment (p-wysocki) [reply tests/model_hub_tests/pytorch/timm_models:539]: Removed.

Comment (p-wysocki) [reply tests/model_hub_tests/pytorch/timm_models:539]: Removed the xfail, added volo_d2 to precommit.

File: src/core/tests/type_prop/col2im.cpp [hunk 2]
```diff
@@ -247,6 +247,49 @@ TEST_F(TypePropCol2ImTest, dynamic_batch_size) {
...
+    const auto output_size = std::make_shared<ShapeOf>(std::make_shared<Parameter>(element::i64, Shape{32, 32}));
+    const auto kernel_size = std::make_shared<ShapeOf>(
+        std::make_shared<Parameter>(element::i64, PartialShape{Dimension::dynamic(), Dimension::dynamic()}));
+    const auto strides = Strides{2, 2};
+    const auto dilations = Strides{2, 2};
+    const auto pads_begin = Shape{3, 3};
+    const auto pads_end = Shape{3, 3};
+
+    const auto op = make_op(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end);
+    op->validate_and_infer_types();
```
Comment (praasz) [inline src/core/tests/type_prop/col2im.cpp:283]: Should not be required.

File: src/core/tests/type_prop/col2im.cpp [hunk 3]
```diff
@@ -247,6 +247,49 @@ TEST_F(TypePropCol2ImTest, dynamic_batch_size) {
...
+    const auto output_size = std::make_shared<ShapeOf>(
+        std::make_shared<Parameter>(element::i64, PartialShape{Dimension::dynamic(), Dimension::dynamic()}));
+    const auto kernel_size = std::make_shared<ov::op::v0::Constant>(element::i64, Shape{2}, std::vector<int32_t>{2, 2});
+    const auto strides = Strides{2, 2};
+    const auto dilations = Strides{2, 2};
+    const auto pads_begin = Shape{3, 3};
+    const auto pads_end = Shape{3, 3};
+
+    const auto op = make_op(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end);
+    op->validate_and_infer_types();
```
Comment (praasz) [inline src/core/tests/type_prop/col2im.cpp:261]: Remove it.

Comment (p-wysocki) [reply src/core/tests/type_prop/col2im.cpp]: Removed.

File: src/core/tests/type_prop/col2im.cpp (hunk 2 above)
Comment (p-wysocki) [reply src/core/tests/type_prop/col2im.cpp]: Removed.

**Review by praasz — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Review by mmikolajcz — APPROVED** [review]

**Review by nshchego — APPROVED** [review]

**Pull request merged**